### ISSUE TITLE
Improve overall performance

### ISSUE THREAD

**Issue by daglu** [digicademy/svsal#38]
General issue with the current app/database, probably needs to be solved through multiple smaller steps (which can be listed here).

**Comment by awagner-mainz** [comment]
Although I am aware of bad experiences when upgrading exist-db, I would like to suggest we check out exist 5 on the test machine. The reason being that besides the usual "performance improvements" that we may or may not profit from, exist 5 should come with a jetty version recent enough to support HTTP/2: https://webtide.com/introduction-to-http2-in-jetty/

**Comment by awagner-mainz** [comment]
Update the following resources:

- [ ] jstree.js
- [ ] bootstrap
- [ ] jquery
- [ ] jquery-ui

Minify the following resources:

- [ ] resources/js/jstree.js
- [ ] resources/js/jquery.tooltipster.js
- [ ] resources/css/mirador-combined.css
- [ ] resources/css/style_work.css
- [ ] resources/css/jquery-ui-1.10.4.custom.css

**Comment by dvglc** [comment]
It might also help if we enhanced access control, especially of functions ('%private'), in the XQuery code, so as to further reduce interdependencies between functions and modules.

**Comment by awagner-mainz** [comment]
These cache settings seem to be good (for eXist-db 4, that is!):
- 6GB total memory (Java -Xmx)
- 1GB Cache Size
- 256MB CollectionCache
- (broker) pool max=100

Smaller Cache resulted in more Cache Misses than Hits, larger resulted in memory mgmt overhead.

**Comment by dvglc** [comment]
Calls to util:eval do not occur in the core of the app, only in services (lod, codesharing).  Hence, we would have to modify "external" code to circumvent such calls. Also, since util:eval only occurs in services, it seems not to play a really important role in the functioning of the app. I would thus refrain from replacing util:eval.

Also, I wouldn't know which other function could replace util:eval, and the same applies to util:expand (which actually plays an important role, e.g. in the tei export, data transformations, etc.). 

Should we leave those functions as they are?

**Comment by awagner-mainz** [comment]
Yes, I think we should leave them as they are now. Possibly the performance hits that we meet from time to time are just one or two blocking queries that are problematic in certain call scenarios, so the main thing would be a good stack trace or other log analysis of such an event.
